ast/compile.go

```diff
@@ -1790,6 +1814,7 @@ func getComprehensionIndex(arity func(Ref) int, candidates VarSet, expr *Expr) *
 	regressionVis := newComprehensionIndexRegressionCheckVisitor(candidates)
 	regressionVis.Walk(body)
 	if regressionVis.worse {
+		dbg.Printf("%s: output vars intersect candiates", expr)
```

Comment:
Nit: `s/candiates/candidates/`. Same filename/line comment applies here and below.